beacon-chain/blockchain/process_block.go

```diff
@@ -550,52 +549,38 @@ func (s *Service) isDataAvailable(ctx context.Context, root [32]byte, signed int
 	if err != nil {
 		return errors.Wrap(err, "could not get KZG commitments")
 	}
-	existingBlobs := len(kzgCommitments)
-	if existingBlobs == 0 {
+	expected := len(kzgCommitments)
+	if expected == 0 {
 		return nil
 	}
 
 	// Read first from db in case we have the blobs
-	s.blobNotifier.Lock()
-	var nc *blobNotifierChan
-	var ok bool
-	nc, ok = s.blobNotifier.chanForRoot[root]
 	sidecars, err := s.cfg.BeaconDB.BlobSidecarsByRoot(ctx, root)
 	if err == nil {
```

Comment:
others errors other than `NotFound` is ignored, we might want to handle the error differently

Reply:
note: this is from previous code

Reply:
This is an outstanding bug, filed as #12851. I think the hangup is that we don't to leave this method because it is triggered by receiving a block via p2p, so we won't re-enter if we return from an error. Perhaps we want to just log db errors here? I'd like to keep that separate to avoid adding other things to this changeset.